salt/orchestrate/edx/init.sls

```diff
@@ -72,24 +80,41 @@ populate_mine_with_edx_node_data:
 reload_pillar_data_on_edx_nodes:
   salt.function:
     - name: saltutil.refresh_pillar
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - require:
         - salt: populate_mine_with_edx_node_data
 
 {# Deploy Consul agent first so that the edx deployment can use provided DNS endpoints #}
 deploy_consul_agent_to_edx_nodes:
   salt.state:
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - sls:
         - consul
         - consul.dns_proxy
 
 build_edx_nodes:
   salt.state:
-    - tgt: 'G@roles:edx and G@environment:{{ ENVIRONMENT }}'
+    - tgt: 'P@roles:(edx|edx-worker) and G@environment:{{ ENVIRONMENT }}'
     - tgt_type: compound
     - highstate: True
     - require:
         - salt: deploy_consul_agent_to_edx_nodes
+
+{% for service in ['edxapp:', 'forum', 'xqueue', 'xqueue_consumer'] %}
+stop_non_edx_worker_services_{{ service }}:
+  salt.function:
+    - name: supervisord.dead
+    - tgt: 'G@roles:edx-worker and G@environment:{{ ENVIRONMENT }}'
+    - tgt_type: compound
+    - kwargs:
+        bin_env: '/edx/bin/supervisorctl'
+        name: {{ service }}
+{% endfor %}
+
+{% for bucket in bucket_prefixes %}
+create_edx_s3_bucket_{{ bucket }}_{{ PURPOSE_PREFIX }}:
+  boto_s3_bucket.create:
+    - name: {{ bucket }}_{{ PURPOSE_PREFIX }}
+{% endfor %}
```

Comment:
This block should go toward the top of the file or add a `require` setting so that the buckets are present by the time the edX instances are launched and configured.